ra_aid/__main__.py

```diff
@@ -166,11 +178,15 @@ def main():
                 style="yellow"
             ))
         
-        # Create the base model after validation
-        model = initialize_llm(args.provider, args.model)
+        # Create the base model after validation, skip if research_only
+        model = None if args.research_only else initialize_llm(args.provider, args.model)
```

Comment:
How come we're setting model to none if we're in research-only mode?

The idea of research-only mode is that it is still a fully agentic flow, it just doesn't make changes to the code base, e.g. we use the read only tool set.

We def want the full agentic workflow with model set, because we want it to be able to do deep research on a codebase to answer questions, which involves exploring the filesystem, calling the read-only research tools, etc.